### PR TITLE
planner: support basic Show in cascades planner

### PR DESCRIPTION
**Pull request by zz-jason**
Signed-off-by: Zhang Jian <zjsariel@gmail.com>

### What problem does this PR solve? 

1. Make it possible to build a physical plan for `LogicalShow` in the cascades planner
2. Refactor some code comments

### What is changed and how it works?

1. Add a new operand for `LogicalShow`.
2. Add an implementation rule for the newly added operand which implements `LogicalShow` to `PhysicalShow`.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

### CODE REVIEW

File: planner/cascades/optimize.go [hunk 1]
```diff
@@ -178,7 +204,13 @@ func onPhaseImplementation(sctx sessionctx.Context, g *memo.Group) (plannercore.
 	return impl.GetPlan(), nil
 }
 
-// implGroup picks one implementation with lowest cost for a Group, which satisfies specified physical property.
+// implGroup finds the best Implementation which the required physical property
```
Comment (francis0407) [inline planner/cascades/optimize.go]:
```suggestion
// implGroup finds the best Implementation which satisfies the required physical property
```

File: planner/cascades/optimize.go [hunk 2]
```diff
@@ -178,7 +204,13 @@ func onPhaseImplementation(sctx sessionctx.Context, g *memo.Group) (plannercore.
...
+// for a Group. The best Implementation should has the lowest cost among all
```
Comment (francis0407) [inline planner/cascades/optimize.go]:
```suggestion
// for a Group. The best Implementation should have the lowest cost among all
```

File: planner/cascades/optimize.go [hunk 3]
```diff
@@ -178,7 +204,13 @@ func onPhaseImplementation(sctx sessionctx.Context, g *memo.Group) (plannercore.
...
+// the applicable Implementations.
+//
+// g:			the Group to be implementated.
```
Comment (francis0407) [inline planner/cascades/optimize.go]:
```suggestion
// g:			the Group to be implemented.
```

**Comment by zz-jason** [comment]
> Any test?

Added. PTAL again, thx.

**Pull request merged**